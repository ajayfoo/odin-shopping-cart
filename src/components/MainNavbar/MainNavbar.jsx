import { Link } from "react-router-dom";
import classes from "./MainNavbar.module.css";
import PropTypes from "prop-types";

export default function MainNavbar({ cartCount }) {
  return (
    <nav className={classes["main-nav"]}>
      <Link to="/">Home</Link>
      <Link aria-label="cart" className={classes.cart} to="/cart">
        Cart
        <span role="alert" aria-label="shopping cart items count">
          {cartCount}
        </span>
      </Link>
    </nav>
  );
}

MainNavbar.propTypes = {
  cartCount: PropTypes.number.isRequired,
};
