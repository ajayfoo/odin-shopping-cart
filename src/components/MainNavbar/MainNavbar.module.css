.main-nav {
  background-color: var(--secondary-container);
  color: var(--on-secondary-container);
  display: flex;
  max-width: fit-content;
  margin: var(--small-gap) auto;
  gap: clamp(4rem, 20vw, 10rem);
  padding: var(--small-gap) var(--medium-gap);
  border-radius: var(--large-radius);
}

.main-nav > a {
  text-decoration: none;
  font-size: var(--medium-font);
}
