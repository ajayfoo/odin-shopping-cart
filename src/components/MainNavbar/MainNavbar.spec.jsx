import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import MainNavbar from "./MainNavbar";
import { MemoryRouter } from "react-router-dom";

describe("Main navigation bar", () => {
  it("renders it", () => {
    render(
      <MemoryRouter>
        <MainNavbar />
      </MemoryRouter>
    );
    screen.getByRole("navigation");

    const home = screen.getByRole("link", { name: "Home", value: "home" });
    const shop = screen.getByRole("link", { name: "Cart", value: "cart" });

    expect(home).toBeVisible();
    expect(home).toHaveTextContent("Home");
    expect(home).toHaveAttribute("href", "/");

    expect(shop).toBeVisible();
    expect(shop).toHaveTextContent("Cart");
    expect(shop).toHaveAttribute("href", "/cart");
  });
});
