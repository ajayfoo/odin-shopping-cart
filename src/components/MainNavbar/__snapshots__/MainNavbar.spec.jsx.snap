// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main navigation bar > renders it 1`] = `
<nav>
  <a
    href="/"
  >
    Home
  </a>
  <a
    href="shop"
  >
    Shop
  </a>
</nav>
`;
